extension/js/download_History.js

```diff
@@ -0,0 +1,114 @@
+var initial = document.body.parentNode.innerHTML;
+
+//open list of already indexed urls from local storage
+var existing_urls = JSON.parse(localStorage['list_downloaded_urls']);
+var abort_status = 0
+
+// activiator in case user cancelles download
+document.getElementById('abort_button').onclick = function(){
+    abort_status = 1
+};
+
+function downloadHistoryUtil(history_items, index) {   
+    //abort_status = localStorage.getItem('abort_status');
+    // Process that runs, as soon as all urls from the lsit have been processed/downloaded
+    if(parseInt(index) === history_items.length) {
+        console.log('Finished Downloading ' + parseInt(index) + ' items');
+
+        // updating list of existing urls and storing it to local storage
+        localStorage.setItem('downloaded_history_items', index);
+        localStorage['list_downloaded_urls'] = JSON.stringify(existing_urls);
+
+        //changing information on downloaded_history.html AFTER succesfully working through list
+        document.getElementById("title_download").innerHTML = "Success! Your History is imported!";
+        document.getElementById("close_message").innerHTML = "You can close this window now.";
+        document.getElementById("info_text").innerHTML = '<p>To activate the search, just type "w" + space + your keyword into the address bar.</p>';
+    }
+
+
+    else {
+
+        var xhttp = new XMLHttpRequest();
+
+        // check if user cancelled download
+        xhttp.onreadystatechange = function() {
+            if (abort_status == 1) {
+                localStorage['list_downloaded_urls'] = JSON.stringify(existing_urls);
+                //localStorage.setItem('abort_status',0)
+                document.getElementById("title_download").innerHTML = "Download cancelled!";
+                document.getElementById("close_message").innerHTML = "You can always restart it via the settings.";
+                document.getElementById("info_text").innerHTML = '<a href="preferences.html">Go to settings</a>';
+                document.getElementById('abort_button').remove()
```

Comment:
at line 8 you have
```
document.getElementById('abort_button').onclick = function(){
    abort_status = 1
};
```
which look like a bind but I just realized is not (never saw this tecnique before), you can just ignore this comment